runtime/lua/vim/lsp/util.lua

```diff
@@ -207,6 +207,7 @@ end
 --@param text_edits (table) list of `TextEdit` objects
 --@param buf_nr (number) Buffer id
 function M.apply_text_edits(text_edits, bufnr)
+  bufnr = bufnr or api.nvim_get_current_buf()
```

Comment:
I noticed that `bufnr` is `nil` when called though `vim.lsp.buf.formatting_sync()`
Most functions accept `0` as the current buffer, but not all functions (like `getmarklist`) accept `nil`